service/src/main/java/uk/nhs/adaptors/gp2gp/ehr/EhrResendController.java

```diff
@@ -0,0 +1,131 @@
+    private static final String CONFLICT = "CONFLICT";
+    private static final String INVALID_IDENTIFIER_VALUE = "INVALID_IDENTIFIER_VALUE";
+
+    private EhrExtractStatusRepository ehrExtractStatusRepository;
+    private TaskDispatcher taskDispatcher;
+    private RandomIdGeneratorService randomIdGeneratorService;
+    private final TimestampService timestampService;
+    private final FhirParseService fhirParseService;
+
+    @PostMapping("/{conversationId}")
+    public ResponseEntity<String> scheduleEhrExtractResend(@PathVariable String conversationId) {
+        EhrExtractStatus ehrExtractStatus = ehrExtractStatusRepository.findByConversationId(conversationId).orElseGet(() -> null);
+
+        if (ehrExtractStatus == null) {
+            var details = getCodeableConcept(INVALID_IDENTIFIER_VALUE);
+            var diagnostics = "Provide a conversationId that exists and retry the operation";
+
+            var operationOutcome = createOperationOutcome(OperationOutcome.IssueType.VALUE,
+                                                          OperationOutcome.IssueSeverity.ERROR,
+                                                          details,
+                                                          diagnostics);
+            var errorBody = fhirParseService.encodeToJson(operationOutcome);
+
+            return new ResponseEntity<>(errorBody, HttpStatus.NOT_FOUND);
+        }
+
+        if (hasNoErrorsInEhrReceivedAcknowledgement(ehrExtractStatus) && ehrExtractStatus.getError() == null) {
+
+            var details = getCodeableConcept(CONFLICT);
+            var diagnostics = "The current resend operation is still in progress. Please wait for it to complete before retrying";
+            var operationOutcome = createOperationOutcome(OperationOutcome.IssueType.BUSINESSRULE,
+                                                          OperationOutcome.IssueSeverity.ERROR,
+                                                          details,
+                                                          diagnostics);
+            var errorBody = fhirParseService.encodeToJson(operationOutcome);
+            return new ResponseEntity<>(errorBody, HttpStatus.CONFLICT);
+        }
+
+        LOGGER.info("Creating tasks to start the EHR Extract process resend");
+        var updatedEhrExtractStatus = prepareEhrExtractStatusForNewResend(ehrExtractStatus);
+        ehrExtractStatusRepository.save(updatedEhrExtractStatus);
+        createGetGpcStructuredTask(updatedEhrExtractStatus);
+
+        return new ResponseEntity<>(HttpStatus.ACCEPTED);
+    }
+
+    private static CodeableConcept getCodeableConcept(String codeableConceptCode) {
+        return new CodeableConcept().addCoding(
+            new Coding("http://fhir.nhs.net/ValueSet/gpconnect-error-or-warning-code-1", codeableConceptCode, null));
+    }
+
+    private static boolean hasNoErrorsInEhrReceivedAcknowledgement(EhrExtractStatus ehrExtractStatus) {
```

Comment:
Minor comment, but this doesn't really feel like it belongs in the Controller. Can refactor this in a later PR though.